### ISSUE TITLE
Sorting Layer Changes when pulled from GitHub.

### ISSUE THREAD

**Issue by dkoester306** [Seanba/Tiled2Unity#103]
I've been working on a project with some friends and we decided to use Tiled and Tiled2Unity for the map. One issue that I'm having is that whenever someone downloads the repository for the first time and opens the map, the sorting order of certain layers are changed. When the project is opened for the first time, GitHub shows that there are at least 3 changes to the map prefab and a couple other things. I've found that undoing these changes and reloading the project fixes this issue for right now. I was wondering if anyone has ever experienced this issue? And if someone might know a fix to stop this, it would be really appreciated! Please leave a response, or if you have questions regarding the post.

**Comment by Seanba** [comment]
I haven't experienced this issue. Are you using "Visible Meta Files" in your editor settings (in Unity)? It sounds to me like collaborators are having to re-export any exported *.tiled2unity.xml files that are in your repository. Also, make sure that everyone is on the same version of Unity. Otherwise, all your content (textures, models, etc.) will be re-imported (and not just the stuff in Tiled2Unity).

This link may be helpful: https://docs.unity3d.com/Manual/Versioncontrolintegration.html

**Comment by dkoester306** [comment]
Thanks for responding to quickly. So I do have "Visible Meta Files" enabled in my editor, so I'm not sure what's causing it. I'm going to include some screenshots of what I'm experiencing in both GitHub and Unity, in case that might help this issue. I'm not sure if this is a problem with my .gitignore or not either.

**What it looks like before the fix.**
![capture](https://cloud.githubusercontent.com/assets/26752085/26602450/1ce856dc-4549-11e7-822c-de8a6b331816.PNG)


**The strange thing is that my third layer "Pond" saves as my "floor" sorting layer but the rest don't.**
![capture4](https://cloud.githubusercontent.com/assets/26752085/26602501/4b3a7a60-4549-11e7-8cc1-1219b0a4d3ee.PNG)
![capture5](https://cloud.githubusercontent.com/assets/26752085/26602503/4b527278-4549-11e7-95f4-0386d3ae760c.PNG)
![capture6](https://cloud.githubusercontent.com/assets/26752085/26602502/4b4c75bc-4549-11e7-99a1-a01c60a4b18c.PNG)

**After exiting unity and discarding these changes, my thing is fixed.**
![capture2](https://cloud.githubusercontent.com/assets/26752085/26602171/309766c4-4548-11e7-85b7-a9be04762a17.PNG)

**What it looks like after the fix, like how it should be.**
![capture7](https://cloud.githubusercontent.com/assets/26752085/26602532/68828ba8-4549-11e7-8362-f5b7023a2895.PNG)

**Comment by Seanba** [comment]
Could you grant me temporary access to your Github? It would help me understand what's going on here a bit better.

**Comment by dkoester306** [comment]
Just did, let me know what you find!

**Comment by Seanba** [comment]
I'm looking at your PlaceHolderMap.tiled2Unity.xml file (this is what Tiled2Unity scripts use to create your prefab) and looks like none of these Tile Layers:

- Ground
- Water
- Pond

... are using the "floor" sorting layer so they revert to "Default". (They do have sorting orders of 0, 1, and 2, respectively, however.)

Also, the map looks right to me when I load the project for the first time ...

![image](https://cloud.githubusercontent.com/assets/6980630/26747295/69c01eb8-47ba-11e7-8a3e-895b1ccf8b52.png)

However, your map prefab in your TestScene scene has "local" changes to it. In other words, someone may have changed the "Pond" sorting layer to "Floor" by hand and then submitted it to your depot. Pressing "Revert" on the PlaceHolderMap game object in the Inspector window undoes these changes and also "fixes" the sorting layers ...

![image](https://cloud.githubusercontent.com/assets/6980630/26747429/4ddaae7e-47bb-11e7-99bf-f1627fee0c4f.png)

Do you have some people on your team on Mac/Unix and others on Windows? It looks like this map was first exported (and submitted) through a OS that uses Mac/Linux line-endings. That is why your material file has local changes on it, btw.

**Comment by dkoester306** [comment]
First off, I'm not sure why GitHub isn't saving the changes to the sorting layers, because I make sure they are changed to the "Floor" sorting layer before I push. 

Also, I'm not sure how you get the map to look like that right when it loads up, because whenever I load the map for the first time it looks like this. 
![image](https://cloud.githubusercontent.com/assets/26752085/26758557/4cd9dbac-48a7-11e7-987d-80e50dd65f04.png)




No one on my team is using Max/Unix, so I'm not sure why the map is exporting like it is for those platforms. At this point I'm not sure if it's just the way I have my unity setup. So, I might try this on a different computer and see if it works any differently. Not sure if there is anything that can be done on your end. I still appreciate all of the help you have provided so far.

**Comment by Seanba** [comment]
When syncing up for the first time your Unity project has build out the "Library" folder. This will also happen when you change build target platforms - like from Windows to iPhone or Playstation 4. This forces first time users to re-import all their resources, including any *.tiled2unity.xml files.

This in turn will recreate the map prefab that you submitted. So, any changes you made to the prefab, like changing the sorting layer, will be undone. If you're making changes to the prefab that Tiled2Unity makes for you then those changes will be lost the next time your re-import that map.

So, what you want to do is have Tiled2Unity automatically set the sorting layer for you when it exports a map. This can be done through the `unity:sortingLayerName` custom property in Tiled (see this link: http://www.seanba.com/megadadadventures.html)

So, long story short, if you set up your map, in Tiled, with the proper attributes then re-export then everything should be okay.

**Comment by dkoester306** [comment]
Hey Seanba,

That worked thank you so much you're a livesaver!! I never would have though to make custom attributes to my Tiled project in order to get it working. Thanks for all your help!!

**Comment by Seanba** [comment]
Glad it worked for you. (And sorry for the late reply)
